crates/cli/tests/reference/namespace.rs

```diff
@@ -0,0 +1,52 @@
+use wasm_bindgen::prelude::*;
+
+/// C-style enum
+#[wasm_bindgen]
+#[derive(Copy, Clone)]
+pub enum ImageFormat {
+    PNG,
+    JPEG,
+    GIF,
+}
+
+#[wasm_bindgen]
+impl ImageFormat {
+    pub fn from_str(s: &str) -> ImageFormat {
+        match s {
+            "PNG" => ImageFormat::PNG,
+            "JPEG" => ImageFormat::JPEG,
+            "GIF" => ImageFormat::GIF,
+            _ => panic!("unknown image format: {}", s),
+        }
+    }
+
+    /// I will be ignored by the generated JS bindings.
+    pub fn is_lossless(self) -> bool {
+        matches!(self, ImageFormat::PNG | ImageFormat::GIF)
+    }
```

Comment:
This should instead be outputted as-is by the proc-macro and not generate any `extern "C" fn` blocks.

Alternatively, instead of going through that work, you can go ahead and implement instance methods as you have suggested.